charmcraft/errors.py

```diff
@@ -171,3 +175,22 @@ def __init__(self, extra_dependencies: Iterable[str]):
 
 class ExtensionError(CraftError):
     """Error related to extension handling."""
+
+
+class SubprocessError(CraftError):
+    """A craft-cli friendly subprocess error."""
+
+    @classmethod
+    def from_subprocess(cls, error: subprocess.CalledProcessError) -> Self:
+        """Convert a CalledProcessError to a craft-cli error."""
+        error_details = f"Full command: {shlex.join(error.cmd)}\nError text:\n"
+        if isinstance(error.stderr, str):
+            error_details += textwrap.indent(error.stderr, "  ")
+        else:
```

Comment:
`stderr` might also be `bytes` right?

Reply:
Yes, but the use case we have for this is always with `text=True`, so it shouldn't be an issue.